src/main/typescript/rug/operations/Decorators.ts

```diff
@@ -17,11 +17,14 @@ function set_metadata(obj: any, key: string, value: any) {
 }
 
 function get_metadata(obj: any, key: string) {
+  if (obj == null) {
+    return null;
+  }
   let desc = Object.getOwnPropertyDescriptor(obj, key);
-  if ((desc == null || desc === undefined) && (obj.prototype !== undefined)) {
-    desc = Object.getOwnPropertyDescriptor(obj.prototype, key);
+  if ((desc == null || desc === undefined) && (Object.getPrototypeOf(obj) !== undefined)) {
+    desc = get_metadata(Object.getPrototypeOf(obj), key);
   }
-  if (desc != null || desc !== undefined) {
+  if (desc != null && desc !== undefined) {
```

Comment:
Since `undefined` == `null`, I don't think this will ever hit the second part of the conditional.